list/List.js

```diff
@@ -186,6 +210,31 @@ define([
 		}),
 
 		/**
+		 * Defines which panel (list | no-items node | none) has to be shown
+		 * depending of the content of the list and id showNoItems is equal to `true` or not.
+		 * @private
+		 */
+		_updateListView: function () {
+			this._displayedPanel = (this._busy) ? "loading-panel" :
+				(this.containerNode && this.containerNode.children.length > 0) ?
+					"list" : ((this.showNoItems) ? "no-items" : "none");
+		},
+
+		////////////////////////////////////////////////////////////////////////////////////////////
+		// Override setAttribute() etc. to put aria-label etc. onto the containerNode rather than the root
+		// node, so that screen readers work properly.
+
+		setAttribute: dcl.superCall(function (sup) {
+			return function (name, value) {
+				if (/^aria-/.test(name) && this.containerNode) {
+					this.containerNode.setAttribute(name, value);
+				} else {
+					sup.call(this, name, value);
+				}
+			};
+		}),
```

Comment:
To be thorough you should override the other methods too (getAttribute(), removeAttibute(), hasAttribute()), like FormWidget does.